docs/intro-to-dune/index.md

```diff
@@ -0,0 +1,284 @@
+root directory of your project. If you've been using the [starter
+project](/installation/#download-and-run-the-starter-project),
+then you've been using Dune this whole time and therefore already have a
+`dune-project` file:
+
+<<< @/../dune-project{clj}
+
+The line
+
+```clj
+(using melange 0.1)
+```
+
+is necessary because we have to manually enable the Melange plugin for Dune in
+order to use it. Note that the version of the Melange Dune plugin is independent
+of the version of Melange we're using.
+
+Technically, `dune-project` accepts many more metadata fields, but it’s best to
+keep it minimal. Other than `name`, it makes more sense to put the rest of your
+project's metadata fields in your `.opam` file, which we'll cover
+[later](/todo/).
+
+`dune-project` files use
+[S-expressions](https://en.wikipedia.org/wiki/S-expression), which might make
+you think of the [Lisp programming
+language](https://en.wikipedia.org/wiki/Lisp_(programming_language)). However,
+S-expressions are just a convenient syntax for encoding structured data, and
+Dune doesn't have the power of a full scripting language.
+
+Each S-expression at the top level is a known as a *stanza*. All the possible
+stanzas you can use in `dune-project` can be found in Dune’s [Stanza
+Reference](https://dune.readthedocs.io/en/stable/dune-files.html#dune-project).
+
+## `dune` files
+
+Besides `dune-project`, Dune also looks at the `dune` files in our project.
+Basically, `dune` files tell Dune about directories, executables, libraries,
+tests, and anything else of interest. For example, here's the `dune` file inside
+the root directory of your project:
+
+<<< @/../dune{clj}
+
+Like `dune-project`, a `dune` file consists of one or more *stanzas*. The only
+stanza here is
+[dirs](https://dune.readthedocs.io/en/stable/dune-files.html#dirs), which tells
+Dune which directories to include in the build. Note that the stanzas accepted
+in `dune` files are not the same as the ones accepted by `dune-project`. See all
+possible `dune` stanzas in Dune's [Stanza
+Reference](https://dune.readthedocs.io/en/stable/dune-files.html#dune).
+
+## `melange.emit` stanza
+
+Let's now take a look at the `src/dune` file:
+
+<<< @/../src/counter/dune{clj}
+
+The main stanza of interest for us is
+[melange.emit](https://dune.readthedocs.io/en/stable/melange.html#melange-emit),
+which tells Dune to turn our OCaml files into JavaScript files. The fields we
+give to `melange.emit` here are `target`, `libraries`, `preprocess`, and
+`module_systems`, which are ones that we need to use for pretty much every
+Melange project.
+
+A note about the `target` field: It is referenced in the `serve` npm script,
+whose command is
+
+```bash
+webpack serve --open --mode development --entry ./_build/default/src/output/src/Index.js
+```
+
+If you change the value of `target` to, say, `stuff`, the value of the `--entry`
+option must be changed to
+
+```
+./_build/default/src/stuff/src/Index.js
+```
```

Comment:
There is a section that explains these paths in the melange docs, maybe worth linking: https://melange.re/v2.1.0/build-system/#javascript-artifacts-layout

Reply:
Added a link to that section in https://github.com/melange-re/melange-for-react-devs/pull/6/commits/e7c52e18e1a9f2c7ca1d191d0be4647c4e6170ae